packages/aws-rfdk/lib/deadline/lib/configure-spot-event-plugin.ts

```diff
@@ -0,0 +1,663 @@
+   */
+  readonly region?: string;
+
+  /**
+   * The length of time that an AWS Worker will wait in a non-rendering state before it is shutdown.
+   * Should evenly divide into minutes.
+   *
+   * @default Duration.minutes(10)
+   */
+  readonly idleShutdown?: Duration;
+
+  /**
+   * Determines if Deadline Spot Event Plugin terminated AWS Workers will be deleted from the Workers Panel on the next House Cleaning cycle.
+   *
+   * @default false
+   */
+  readonly deleteSEPTerminatedWorkers?: boolean;
+
+  /**
+   * Determines if EC2 Spot interrupted AWS Workers will be deleted from the Workers Panel on the next House Cleaning cycle.
+   *
+   * @default false
+   */
+  readonly deleteEC2SpotInterruptedWorkers?: boolean;
+
+  /**
+   * Determines if any active instances greater than the target capacity for each group will be terminated.
+   * Workers may be terminated even while rendering.
+   *
+   * @default false
+   */
+  readonly strictHardCap?: boolean;
+
+  /**
+   * The Spot Event Plugin will request this maximum number of instances per House Cleaning cycle.
+   *
+   * @default 50
+   */
+  readonly maximumInstancesStartedPerCycle?: number;
+
+  /**
+   * Determines how the Spot Event Plugin should handle Pre Job Tasks.
+   * See https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/job-scripts.html
+   *
+   * @default SpotEventPluginPreJobTaskMode.CONSERVATIVE
+   */
+  readonly preJobTaskMode?: SpotEventPluginPreJobTaskMode;
+
+  /**
+   * The Worker Extra Info column to be used to display AWS Instance Status
+   * if the instance has been marked to be stopped or terminated by EC2 or Spot Event Plugin.
+   * All timestamps are displayed in UTC format.
+   *
+   * @default SpotEventPluginDisplayInstanceStatus.DISABLED
+   */
+  readonly awsInstanceStatus?: SpotEventPluginDisplayInstanceStatus;
+}
+
+/**
+ *  Connection options used if render queue has TLS enabled.
+ */
+export interface ConfigureSpotEventPluginTrafficEncryptionProps {
+  /**
+   * The certificate used to sign the the chain of trust used for render queue.
+   */
+  readonly caCert: ISecret;
+}
+
+/**
+ * Input properties for ConfigureSpotEventPlugin.
+ */
+export interface ConfigureSpotEventPluginProps {
+  /**
+   * The VPC in which to create the network endpoint for the lambda function that is
+   * created by this construct.
+   */
+  readonly vpc: IVpc;
+
+  /**
+   * The RenderQueue that Worker fleet should connect to.
+   */
+  readonly renderQueue: IRenderQueue;
+
+  /**
+   * Where within the VPC to place the lambda function's endpoint.
+   *
+   * @default The instance is placed within a Private subnet.
+   */
+  readonly vpcSubnets?: SubnetSelection;
+
+  /**
+   * Connection options if the network traffic to the RenderQueue should is encrypted.
+   */
+  readonly trafficEncryption?: ConfigureSpotEventPluginTrafficEncryptionProps;
```

Comment:
We should be able to get this straight from the RenderQueue object, no?

The [endpoint property](https://docs.aws.amazon.com/rfdk/api/latest/docs/aws-rfdk.deadline.RenderQueue.html#endpoint) would be able to tell us whether we're connecting via TLS or not, and which port we connect on. We also have a property on the RenderQueue for the client [certificate chain](https://github.com/aws/aws-rfdk/blob/e6bb60dc5cb186dca16b70daf4990e845b2825e1/packages/aws-rfdk/lib/deadline/lib/render-queue.ts#L226), though it's currently private. Seems reasonable to change it to a public property and, maybe, rename it if it's not clear enough what it's for.

That would remove a potential pitfall if a customer were to configure the RenderQueue for TLS, but accidentally forget to put the option here.

Reply:
Looks much better now, thanks!